lightning_examples/fsd50K-speech-model-finetuning/fsd50K_speech_model_finetuning.py

```diff
@@ -0,0 +1,271 @@
+            "hidden_size": 512,
+            "normalization": nn.BatchNorm1d,
+        },
+        "loss_cls": nn.BCEWithLogitsLoss,
+        "loss_prm": {},
+        "optimizer_cls": Adam,
+        "optimizer_prm": {
+            "lr": 1e-5,
+        },
+        "unfreeze": {
+            "encoder.layers.11": 1,
+        },
+    },
+    "trainer_config": {
+        "max_epochs": 3,
+        "auto_select_gpus": True,
+        "accelerator": "gpu",
+        "devices": 1,
+        "check_val_every_n_epoch": 1,
+        "precision": 16,
+        "callbacks": [
+            pl.callbacks.ModelCheckpoint(
+                filename="{epoch}-{val_loss:.5f}",
+                save_top_k=-1,
+                monitor="val_loss",
+                mode="min",
+            ),
+            Unfreeze(),
+        ],
+        "logger": pl.loggers.TensorBoardLogger(
+            save_dir="tb_logs",
+        ),
+    },
+}
+
+# %% [markdown] id="dUGV1hdbVsUV"
+# ## Set seed and add params deduced from user's configuration
+
+# %% id="uNNvu1Jd5kIH"
+pl.utilities.seed.seed_everything(FULL_CONFIG["seed"])
+FULL_CONFIG["trainer_config"]["log_every_n_steps"] = (
+    LOG_INTERVAL_SAMPLES // FULL_CONFIG["datamodule_config"]["batch_size"]
+)
+
+# %% [markdown] id="daXR-HHz4ze7"
+# # Prepare (reduced) dataset
+
+# %% [markdown] id="KZ9vf9o4UWEg"
+# ## Download and unzip
+
+# %% id="pqQlayizo8YX"
+# !gdown 1cOcOEK56p6k2RNbM-10QFHOD4jenqHym -O /content/
+# !unzip './FSD50K_500.zip' -d './dataset'
+
+# %% [markdown] id="W-vcoLY1RqxH"
+# ## Inspect data
+
+# %% id="S89t3R0zi0nj"
+fsd50k_datadict = FSD50KDataDict(**FULL_CONFIG["datamodule_config"]["datadict_prm"])
+
+# %% id="F1BlO17ViNew"
+train_datadict = fsd50k_datadict.get_dict("train")
+
+# %% [markdown] id="N4SIRhKjgqw8"
+# Add a dict entry containing the labels as strings for inspection.
+
+# %% id="LiuSKBUThw1O"
+train_datadict["ys_true_names"] = tokens_to_names(train_datadict["ys_true"], fsd50k_datadict.token_to_name)
+
+# %% id="UwDU12Ewj-fA"
+inspect_data(
+    datadict=train_datadict,
+    show_keys=["paths", "ys_true_names"],
+    samples_indices=range(5),
+)
+
+# %% [markdown] id="wVe_yrk54Axz"
+# # Launch fine-tuning
+
+# %% id="BP7HBa7C87IY"
+# Inside an interactive environment, logs could be observed using:
+# # %load_ext tensorboard
+# # %tensorboard --logdir ./tb_logs
+
+# %% id="dZl2TdlTxUkF"
+model = EmbedderClassifier(**FULL_CONFIG["model_config"])
+
+datamodule = FSD50KDataModule(**FULL_CONFIG["datamodule_config"])
+
+trainer = pl.Trainer(**FULL_CONFIG["trainer_config"])
+
+trainer.fit(model, datamodule=datamodule)
+
+# %% [markdown] id="jzVXKW-L4E2R"
+# # Evaluate performance
+
+# %% [markdown] id="bPbyVevnZXTX"
+# ## Predict on test data
+
+# %% id="6VTZHQDw2y5p"
+preds = trainer.predict(ckpt_path="best", datamodule=datamodule)
+
+# %% id="FM0fc3UbSp-2"
+preds = gather_preds(preds)
+
+# %% [markdown] id="cD556ynvAbYa"
+# ## Compute metrics
+
+# %% id="zlTooqqp8FWk"
+mAP_micro = average_precision_score(

```

Comment:
What do you mean? Something like this?

With `on_step=False, on_epoch=True` to only log the end of the epoch according to https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html#logging-from-a-lightningmodule:
> The above config for validation applies for test hooks as well.

```suggestion
# In __init__:
self.mAP = torchmetrics.classification.MultilabelAveragePrecision()

# In test_step:
self.mAP(preds, y)
self.log('mAP', self.mAP)
```

Reply:
Then the class version of torchmetrics should be prefered to functional I'd say?

Reply:
okay.. it's fine.. let's use functional metrics here since you already have all the targets and predictions.
modular metrics are useful, when you are aggregating the metrics, let's say on step-level